### DISCUSSION TITLE
What is the equivalent WHERE of C++ version in Rust version?

### DISCUSSION THREAD

**Discussion by 3togo** [arrayfire/arrayfire-rust#334] [Q&A] [answered]
I note that there are many functions missing in arrayfire-rust when compared with arrayfire_c++.
1. where
AFAPI array af::where 	( 	const array &  	in	) 	
C++ Interface for finding the locations of non-zero values in an array. 
2.  assignment
    array reference = A(span, 2);
    A(span, 2) = 3.14f;
    assert(allTrue<bool>(reference != A(span, 2)));

How can I do assignment and do searching by where in arrayfire-rust?

**Comment by 9prady9** [comment, marked as answer]
since where is a keyword in rust, we have https://arrayfire.org/arrayfire-rust/arrayfire/fn.locate.html here. please refer to this tutorial for indexing and assignments https://arrayfire.org/arrayfire-rust/book/indexing.html

**Reply by 3togo** [reply]
I want to write something like below.  Is there any examples?
 let x = a > 0.5;
 let indices = locate(&x);
 let x_filtered = x[indices];

**Reply by 9prady9** [reply]
Please look at eval macro - examples from the indexing tutorial should give you an idea how to achieve the above.

**Reply by 3togo** [reply]
@9prady9 

many thanks

Eli
